week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,39 @@
-function setAlarm() {}
+function setAlarm() {
+  let count = document.getElementById("timeRemaining");
+  let setIt = document.getElementById("set");
+  let pause = document.getElementById("p");
+  let Restart = document.getElementById("re");
+  let Counter;
+  pause.addEventListener("click", function () {
+    clearInterval(Counter);
+  });
+  Restart.addEventListener("click", function () {
+    Counter = setInterval(CountDown, 1000);
+  });
+  let TheTime;
+  setIt.addEventListener("click", function () {
+    TheTime = document.getElementById("alarmSet").value;
+
+    if (TheTime <= 60) {
+      count.innerHTML = "Time Remaining: 00:" + TheTime;
+    }
+  });
+  Counter = setInterval(CountDown, 1000);
+  function CountDown() {
+    if (TheTime <= 60) {
+      count.innerHTML = "Time Remaining: 00:" + TheTime;
+    }
```

Comment:
This part is duplicated. Do you remember the DRY rule? Meaning Do not Repeat Yourself. When you have repeated code lines, it is worth considering to factor those out into a separate function and just you have to refer to the function.